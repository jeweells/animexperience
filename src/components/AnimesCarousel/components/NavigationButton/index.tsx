import React from "react";
import { Button, Icon } from "rsuite";
import styled from "styled-components";

const NavContainer = styled.div`
    position: absolute;
    top: 0;
    bottom: 0;
    z-index: 10;
`;

export type NavigationButtonProps = {
    direction: "right" | "left";
    onClick?(): void;
    disabled?: boolean;
}

const SButton = styled(Button)<Pick<NavigationButtonProps, "direction">>`
    height: 100%;
    background-color: transparent !important;
    opacity: ${props => props.disabled ? 0 : 0.7} !important;
    background-image: linear-gradient(to ${
    props => props.direction}, rgba(0,0,0,0), rgba(0,0,0,.3) 50%, rgba(0,0,0,.4));
    transition: all 300ms ease-in-out;
    padding: 0 24px;
    &:hover {
        opacity: ${props => props.disabled ? 0 : 1} !important;
        background-image: linear-gradient(to ${
    props => props.direction}, rgba(0,0,0,0), rgba(0,0,0,.4) 50%,  rgba(0,0,0,.8));
    }
`;

export const NavigationButton = React.memo(React.forwardRef<HTMLDivElement, NavigationButtonProps>(({
    direction,
    onClick,
    disabled,
}, ref) => {
    return (
        <NavContainer
            ref={ref}
            style={{ [direction]: 0 }}
        >
            <SButton
                direction={direction}
                disabled={disabled}
                onClick={onClick}
            >
                <Icon icon={`angle-${direction}` as any} size={"5x"}/>
            </SButton>
        </NavContainer>
    );
}));

NavigationButton.displayName = "NavigationButton";

export default NavigationButton;
