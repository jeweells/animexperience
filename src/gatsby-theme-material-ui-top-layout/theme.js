import { createTheme } from '@material-ui/core/styles'

export const mainColor = 'rgb(105, 190, 213)'

const theme = createTheme({
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    overrides: {
        MuiButtonBase: {
            root: {
                borderRadius: undefined,
            },
        },
        MuiButton: {
            root: {
                color: '#e9ebf0',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                height: 36,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },

            text: {
                color: '#e9ebf0',
                textTransform: 'none',
                padding: '8px 12px',
            },
        },
        MuiIconButton: {
            root: {
                flexShrink: 0,
                height: 36,
                minWidth: 36,
                color: '#e9ebf0',
                padding: '8px 12px',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },
        },
        MuiCssBaseline: {
            '@global': {
                html: {
                    WebkitFontSmoothing: 'auto',
                },
                body: {
                    color: undefined,
                    margin: undefined,
                    fontSize: undefined,
                    fontFamily: undefined,
                    fontWeight: undefined,
                    backgroundColor: undefined,
                    lineHeight: undefined,
                },
            },
        },
        MuiSkeleton: {
            root: {
                backgroundColor: 'rgba(255, 255, 255, 0.11)',
            },
            pulse: {
                animation: 'MuiSkeleton-keyframes-pulse 3s ease-in 0.5s infinite',
            },
        },
    },
})

export default theme
