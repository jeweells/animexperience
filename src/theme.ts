import { createTheme } from '@mui/material/styles'

const theme = createTheme({
    breakpoints: {
        values: {
            xs: 0,
            sm: 500,
            md: 800,
            lg: 1000,
            xl: 1200,
        },
    },
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        mode: 'dark',
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    components: {
        MuiSvgIcon: {
            styleOverrides: {
                root: {
                    fontSize: 'inherit',
                },
            },
        },
        MuiButtonBase: {
            styleOverrides: {
                root: {
                    borderRadius: undefined,
                    textTransform: 'unset',
                },
            },
        },
        MuiButton: {
            styleOverrides: {
                root: {
                    color: '#e9ebf0',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: 14,
                    lineHeight: '1.42857143',
                    textTransform: 'unset',
                    borderRadius: 6,
                    height: 36,
                    '&:hover': {
                        outline: 'none',
                        border: 'none',
                        backgroundColor: '#3c3f43',
                    },
                },
                text: {
                    color: '#e9ebf0',
                    textTransform: 'none',
                    padding: '8px 12px',
                },
            },
        },
        MuiIconButton: {
            styleOverrides: {
                root: {
                    flexShrink: 0,
                    height: 36,
                    minWidth: 36,
                    color: '#e9ebf0',
                    padding: '8px 6px',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: '1.2rem',
                    lineHeight: '1.42857143',
                    borderRadius: 6,
                    '&:hover': {
                        backgroundColor: '#3c3f43',
                    },
                    '&.Mui-disabled': {
                        color: 'rgb(255 255 255 / 25%)',
                    },
                },
            },
        },
        MuiSkeleton: {
            defaultProps: {
                animation: 'pulse',
            },
            styleOverrides: {
                root: {
                    backgroundColor: 'rgba(255, 255, 255, 0.11)',
                },
                pulse: {
                    animationDuration: '1.5s 0.5s',
                },
            },
        },
        MuiCssBaseline: {
            styleOverrides: `
                * {
                    backface-visibility: hidden;
                    margin: 0;
                    padding: 0;
                    box-sizing: border-box;
                }
                html {
                    -webkit-font-smoothing: auto;
                }
                h6 {
                    margin: 0;
                }
                body {
                    background-color: #0F131A;
                }
                ::-webkit-scrollbar {
                    width: 10px;
                }
                ::-webkit-scrollbar-track {
                    background: transparent;
                }
                ::-webkit-scrollbar-thumb {
                    background: #25282e;
                },
                ::-webkit-scrollbar-thumb:hover {
                    background: #393a44;
                }
            `,
        },
    },
})

export default theme
