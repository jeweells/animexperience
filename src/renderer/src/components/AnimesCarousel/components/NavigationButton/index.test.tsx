import { ThemeProvider } from '@mui/material/styles'
import NavigationButton, { NavigationButtonProps } from './index'
import { NAVIGATION_BUTTON } from '@selectors'
import theme from '../../../../theme'

describe.each([['right'], ['left']])('NavigationButton (%s)', (direction) => {
  let props: NavigationButtonProps
  const onClick = jest.fn()

  const getComponent = (override: Partial<NavigationButtonProps> = {}) => {
    return (
      <ThemeProvider theme={theme}>
        <NavigationButton {...props} {...override} />
      </ThemeProvider>
    )
  }

  beforeEach(() => {
    props = {
      direction: direction as NavigationButtonProps['direction'],
      onClick
    }
    onClick.mockReset()
  })

  it('renders default', () => {
    const wrapper = render(getComponent())
    fireEvent.click(wrapper.getByTestId(NAVIGATION_BUTTON.BUTTON))
    expect(onClick).toHaveBeenCalledTimes(1)
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders default when disabled', () => {
    const wrapper = render(getComponent({ disabled: true }))
    fireEvent.click(wrapper.getByTestId(NAVIGATION_BUTTON.BUTTON))
    expect(onClick).toHaveBeenCalledTimes(0)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})
