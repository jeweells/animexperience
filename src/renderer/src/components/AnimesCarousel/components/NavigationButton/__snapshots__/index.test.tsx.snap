// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton (left) renders default 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0.7!important;
  background-image: linear-gradient(
      to left,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-1:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1:hover {
  opacity: 1!important;
  background-image: linear-gradient(
        to left,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.emotion-3 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      class="emotion-0"
      style="left: 0px; width: 72px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-1"
        data-testid="navigation-button-button"
        direction="left"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
          data-testid="KeyboardArrowLeftRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.71 15.88 10.83 12l3.88-3.88c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.39.39-1.03 0-1.42"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-3"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`NavigationButton (left) renders default when disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0!important;
  background-image: linear-gradient(
      to left,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-1:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1:hover {
  opacity: 0!important;
  background-image: linear-gradient(
        to left,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

<body>
  <div>
    <div
      class="emotion-0"
      style="left: 0px; width: 72px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-1"
        data-testid="navigation-button-button"
        direction="left"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
          data-testid="KeyboardArrowLeftRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.71 15.88 10.83 12l3.88-3.88c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.39.39-1.03 0-1.42"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;

exports[`NavigationButton (right) renders default 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0.7!important;
  background-image: linear-gradient(
      to right,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-1:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1:hover {
  opacity: 1!important;
  background-image: linear-gradient(
        to right,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.emotion-3 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      class="emotion-0"
      style="right: 0px; width: 72px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-1"
        data-testid="navigation-button-button"
        direction="right"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
          data-testid="KeyboardArrowRightRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.29 15.88 13.17 12 9.29 8.12a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-3"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`NavigationButton (right) renders default when disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0!important;
  background-image: linear-gradient(
      to right,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-1:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1:hover {
  opacity: 0!important;
  background-image: linear-gradient(
        to right,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

<body>
  <div>
    <div
      class="emotion-0"
      style="right: 0px; width: 72px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-1"
        data-testid="navigation-button-button"
        direction="right"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
          data-testid="KeyboardArrowRightRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.29 15.88 13.17 12 9.29 8.12a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;
